packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,253 @@
+  `,
+
+  xsmall: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 418px;
+  `,
+
+  small: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 562px;
+  `,
+
+  default: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 700px;
+  `,
+
+  large: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 800px;
+  `,
+
+  xlarge: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 1270px;
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: aboslute;
+  float: right;
+  cursor: pointer;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'default', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  shouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = () => {
+    // Don't close modal if shouldClose returns false or no setActive callback is passed
+    if (!setActive || (shouldClose && !shouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  };
+
+  const handleDocumentClick = (e: React.SyntheticEvent) => {
+    if (!contentRef.current) {
+      return;
+    }
+
+    if (!contentRef.current.contains(e.target as Node)) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    if (e.keyCode === escapeKey) {
+      handleClose();
+    }
+  }, []);
```

Comment:
If the consuming component passes in a new `setActive` function for some reason, this callback is going to be stale.

Reply:
I think I'm missing what you are getting at here and with the comment below?

Reply:
The comment below is unrelated (just pointing out that the name of the prop doesn't seem to match what it's called in prop types).

What I'm saying here is related to [this problem](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) outlined in the hooks FAQ.

Reply:
We used this hook with the Menu component, but also seems like it's okay because we aren't using information from the component's scope. The reason we're using the [] is just to ensure it's only called once. Let me know if this makes sense -- definitely open to updating as needed, just not sure I understand the problem yet.

Reply:
Got it!! THANK YOU

Reply:
So do we need to append `setActive` here? I'm unclear if we do.

Reply:
we need handleClose there and setActive in handleClose, both have been added